docs/development/synapse_architecture/faster_joins.md

```diff
@@ -0,0 +1,276 @@
+# How do faster joins work?
+
+This is a work-in-progress set of notes with two goals:
+- act as a reference, explaining how Synapse implements faster joins; and
+- record the rationale behind our choices.
+
+See also [MSC3902](https://github.com/matrix-org/matrix-spec-proposals/pull/3902).
+
+## Overview: processing events in a partially-joined room
+
+The response to a partial join consists of
```

Comment:
Might want to just clarify really quickly here that a partial join is also a fast join/lazy-load join? Sometimes the interchangeable use of terminology can confuse those new to the subject.

Reply:
Will say something to this effect.

I'll try to avoid mentioning lazy-loading. We _are_ doing lazy loading of state, but there's an existing lazy-loading mechanism involved with  /sync. (Indeed, we need to use lazy-loading-of-members in /sync to be able to take of faster joins at _all_ as a client.